### PR TITLE
Fixes to bioframe.select query interval semantics

### LINKED ISSUE #145: select recently changed behavior causing issues in cooltools
as described in https://github.com/open2c/cooltools/issues/435 :

bioframe select became more inclusive and is selecting "more" with v0.4.0 ...
with bioframe 0.3.1 
``` python
In [16]: bioframe.select(track, ("chr1",100000000,150000000)).shape
Out[16]: (50, 4)
```
and with bioframe 0.4.0:
```python
In [4]: bioframe.select(track, ("chr1",100000000,150000000)).shape
Out[4]: (51, 4)
```

sample output of the `bioframe.select` here:
image
that *99th* bins is not included with the older bioframe

should cooltools adapt to this change or will there be an option to do `inclusive/exclusive` selection ? like `inclusive_left`, `inclusive_right` ?

### PR DESCRIPTION
**Pull request by nvictus**
Fixes #145 

* <s>Somewhat hacky solution to dealing with dataframes of genomic points.</s>
* Also fixes incorrect results when end is `None` or implicit `"chr2:10-"`

Update: New query mask ensures inclusion in the edge case where a point interval sits exactly at the query interval start coordinate. As a side-effect, query intervals can also be points and will collect any coincident points in the dataframe, so the behavior appears to be self-consistent.

### CODE REVIEW

**Comment by nvictus** [comment]
Added various tests, but could still use a bit more exploration of potential edge cases we've missed or inconsistencies.

File: bioframe/ops.py [hunk 1]
```diff
@@ -54,12 +54,19 @@ def select_mask(df, region, cols=None):
...
     if chrom is None:
         raise ValueError("no chromosome detected, check region input")
-    if (start is not None) and (end is not None):
-        mask = (df[ck] == chrom) & (df[sk] < end) & (df[ek] >= start)
-    else:
+
+    if start is None:
         mask = df[ck] == chrom
+    else:
+        if end is None:
```
Comment (gfudenberg) [inline bioframe/ops.py:64]: when is start not None but end is None?

**Review by gfudenberg — APPROVED** [review]
nicely readable tests!

File: bioframe/ops.py (hunk 1 above)
Comment (nvictus) [reply bioframe/ops.py:64]: Yup, `bioframe.parse_region` now accepts unspecified ends like `"chr2:100-"`

**Comment by gfudenberg** [comment]
funky!
was this a common use case for you?

**Comment by nvictus** [comment]
It figured it was annoying to always have to look up a chromsizes to specify an `end` for bioframe. And it aligns with Python slice semantics: `x[3:] == x[slice(3, None)]` --> `parse_region("chrX:3-") == ('chrX', 3, None)`

**Comment by gfudenberg** [comment]
Gotcha.
Should (is?)

chrX:-3 supported? I.e. ('chrX',none,3)

**Comment by nvictus** [comment]
Nope. It's messy because it looks like a negative coordinate and it's easy enough to do `chrX:0-3` because the start is known.

**Comment by gfudenberg** [comment]
right, of course!
